src/Microsoft.AspNetCore.Blazor.Razor.Extensions/BlazorImportProjectFeature.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.AspNetCore.Razor.Language;
+
+namespace Microsoft.AspNetCore.Blazor.Razor
+{
+    internal class BlazorImportProjectFeature : IImportProjectFeature
+    {
+        private const string ImportsFileName = "_ViewImports.cshtml";
+
+        public RazorProjectItem DefaultImports => VirtualProjectItem.Instance;
+
+        public RazorProjectEngine ProjectEngine { get; set; }
+
+        public IReadOnlyList<RazorProjectItem> GetImports(RazorProjectItem projectItem)
+        {
+            if (projectItem == null)
+            {
+                throw new ArgumentNullException(nameof(projectItem));
+            }
+
+            var imports = new List<RazorProjectItem>()
+            {
+                VirtualProjectItem.Instance,
+            };
+
+            // We add hierarchical imports second so any default directive imports can be overridden.
+            imports.AddRange(GetHierarchicalImports(ProjectEngine.FileSystem, projectItem));
+
+            return imports;
+        }
+
+        // Temporary API until we fully convert to RazorProjectEngine
+        public IEnumerable<RazorProjectItem> GetHierarchicalImports(RazorProject project, RazorProjectItem projectItem)
+        {
+            // We want items in descending order. FindHierarchicalItems returns items in ascending order.
+            return project.FindHierarchicalItems(projectItem.FilePath, ImportsFileName).Reverse();
+        }
+
+        private class VirtualProjectItem : RazorProjectItem
+        {
+            private readonly byte[] _defaultImportBytes;
+
+            private VirtualProjectItem()
+            {
+                var preamble = Encoding.UTF8.GetPreamble();
+                var content = @"
+@using System
+@using System.Collections.Generic
+@using System.Linq
+@using System.Threading.Tasks
```

Comment:
Let me know if you want me to add anything else here.

Reply:
This can have any non-block directives - ie: no code, only declarations.

Reply:
Great. I'm sure we will want a couple of the most commonly used Blazor namespaces here too, but we can work out exactly which ones later.

Reply:
BTW the strategy we take in MVC to dump all of the names applications are likely to use into 1-2 namespaces so that the strategy is straightforward. For MVC views that's `M.A.Mvc` and `M.A.Mvc.Rendering`.

Also, fully qualify everything we use as a codegen target, so that there's no potential ambiguity.

Mentioning it because I've seen 3-4 different namespaces in tests.